### ISSUE TITLE
New install in a Canvas Beta site is changing Canvas assignment names when the tool is selected

### ISSUE THREAD

**Issue by mkdir-washington-edu** [hypothesis/product-backlog#1458]
### Steps to reproduce
I can't reproduce this in our Prod instance. I watched this happen with a partner in their Beta instance.

In case it matters, prior to the issue we installed in Beta using a number of steps so that we could replicate exactly what they have done in the Prod instance. These steps were
1. Create a new Canvas API key
2. Create a new Canvas 1.1 Key/Secret and install - app id = 5787
3. Create a new Canvas LTI 1.3 Key
4. Create a new deployment at the account level - app id = 5788
5. Edit the 1.1 Placements so the tool does not appear in the app list but is still installed
6. Delete the 1.1 install and the 1.3 deployment
7. Use the API and LTI 1.3 Key to create a new subaccount deployment - app id = 5789

Next, here were the steps to reproduce the issue where the Assignment name in Canvas changes
1. Create a new assignment
2. give it a unique tile and description
3. submission type = external tool
4. choose Hypothesis from the list
5. complete the Hypothesis assignment creation workflow
6. In the Canvas external tool window choose "Select"
7. The Canvas Assignment name changes from the unique name to "HYPOTHESIS". The description does not change.

If you edit the new title and then save the assignment the new title is saved. The partner is concerned that the new install of our app may be causing this issue.

This issue happened with app instance 5789 around 3:00pm ET on Friday

**Comment by mkdir-washington-edu** [comment]
To edit placements in 1.1 to hide an app:

For any 1.1 install, you can go to the app settings page, click the app’s gear icon dropdown, click on Placements, then turn them on or off:
![Image](https://github.com/hypothesis/product-backlog/assets/22603275/7d8e78f0-de53-4a71-a424-38ed42c81b41)
![Image](https://github.com/hypothesis/product-backlog/assets/22603275/25fe3295-362d-4f49-8a4b-c818ffb7fc97)

**Comment by marcospri** [comment]
After a fair amount of investigation I reckon this is a recent behavior change in Canvas, see the issue I created on their tracker: https://github.com/instructure/canvas-lms/issues/2211.

Until we get an answer/solution from them I've tried to find a workaround without any luck:

- Try to get the title from the address bar. We run on an iframe we don't have access to that value due to CORS protections.

- We don't get the assignment title on the deep linking message either.



To test in our own instance head over: https://hypothesis.beta.instructure.com/courses/125/assignments/4682/edit?name=Testing+creation+-+marcos&due_at=null&points_possible=0




See: https://hypothes-is.slack.com/archives/C2BLQDKHA/p1684752881066189
